### PR TITLE
Feature req

### PR DESCRIPTION
**Pull request by ArmantasPik**
-Added better sorting
-Added multithreading for calculating discounts
-Added dependency injection for food items

### CODE REVIEW

File: Models/Food.cs [hunk 1]
```diff
@@ -32,6 +32,8 @@ public int Quantity
             }
         }
         public string FoodCategory { get; set; }
+        public string RestaurantName { get; set; }
+        public int Discount { get; set; }
```
Comment (medikornov) [inline Models/Food.cs:36]: good update

File: Models/SQLFoodRepository.cs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using FoodKept.Data;
+
+namespace FoodKept.Models
+{
+    public class SQLFoodRepository : IFoodRepository
+    {
```
Comment (medikornov) [inline Models/SQLFoodRepository.cs:13]: useful thing

File: Pages/FoodCustomer/Cart.cshtml.cs [hunk 3]
```diff
@@ -169,11 +175,32 @@ public async Task<IActionResult> OnPostRemoveFromCartAsync(int id)
             {
                 smtpClient.Send(message: mail);
             }
-            catch (Exception)
+            catch (SmtpFailedRecipientsException ex)
             {
+                SmtpStatusCode status = ex.InnerExceptions[0].StatusCode;
+
```
Comment (medikornov) [inline Pages/FoodCustomer/Cart.cshtml.cs:181]: thanks for fixing it

**Review by medikornov — APPROVED** [review]

File: Pages/FoodCustomer/Food.cshtml.cs [hunk 4]
```diff
@@ -90,52 +85,26 @@ public async Task OnGetAsync(string sortOrder, string currentFilter, string sear
...
-                case "Category":
-                    foodIQ = foodIQ.OrderBy(s => s.FoodCategory);
-                    break;
-                case "category_desc":
-                    foodIQ = foodIQ.OrderByDescending(s => s.FoodCategory);
-                    break;
-                default:
-                    foodIQ = foodIQ.OrderBy(s => s.FoodName);
-                    break;
+                foodIQ = foodSort.Value.SortCommandHandler[sortOrder](sortOrder.Substring(1), foodIQ);
```
Comment (0rl1s) [inline Pages/FoodCustomer/Food.cshtml.cs:93]: Nice change. Looks much better now :)

File: Pages/FoodCustomer/Food.cshtml.cs [hunk 5]
```diff
@@ -46,9 +45,6 @@ public FoodModel(FoodKept.Data.ShopContext context, UserManager<ApplicationUser>
 
         public async Task OnGetAsync(string sortOrder, string currentFilter, string searchString, string currentCategory, int? pageIndex)
         {
-            //Lazy Initialization
-            //Lazy<List<Food>> getFood = new Lazy<List<Food>>(() => _context.FoodData.ToList());
-
```
Comment (0rl1s) [inline Pages/FoodCustomer/Food.cshtml.cs:51]: Good job for trying to keep code clean and deleting useless comments.

File: Pages/FoodCustomer/Cart.cshtml.cs [hunk 6]
```diff
@@ -169,11 +175,32 @@ public async Task<IActionResult> OnPostRemoveFromCartAsync(int id)
...
+                if (status == SmtpStatusCode.MailboxBusy || status == SmtpStatusCode.MailboxUnavailable)
+                {
+                    ViewData["Message"] = string.Format("Delivery failed - retrying again");
+                    Thread.Sleep(1000);
+                    smtpClient.Send(mail);
```
Comment (0rl1s) [inline Pages/FoodCustomer/Cart.cshtml.cs:186]: Dealing with exceptions in a meaningful way now. Good job.

**Review by 0rl1s — APPROVED** [review]

**Pull request merged**